aw-sync/src/sync.rs

```diff
@@ -85,134 +109,258 @@ impl AccessMethod for AwClient {
         end: Option<DateTime<Utc>>,
         limit: Option<u64>,
     ) -> Result<Vec<Event>, String> {
-        Ok(self.get_events(bucket_id).unwrap())
+        Ok(self.get_events(bucket_id, start, end, limit).unwrap())
     }
-    fn insert_events(&self, bucket_id: &str, events: Vec<Event>) -> Result<Vec<Event>, String> {
-        //Ok(self.insert_events(bucket_id, &events[..]).unwrap())
-        Err("Not implemented".to_string())
+    fn insert_events(&self, bucket_id: &str, events: Vec<Event>) -> Result<(), String> {
+        self.insert_events(bucket_id, events).unwrap();
+        Ok(())
     }
-    fn get_event_count(&self, bucket_id: &str) -> Result<i64, String> {
-        //Ok(self.get_event_count(bucket_id, None, None).unwrap())
-        Err("Not implemented".to_string())
+    fn get_event_count(
+        &self,
+        bucket_id: &str,
+        start: Option<DateTime<Utc>>,
+        end: Option<DateTime<Utc>>,
+    ) -> Result<i64, String> {
+        Ok(self.get_event_count(bucket_id, start, end).unwrap())
     }
     fn create_bucket(&self, bucket: &Bucket) -> Result<(), DatastoreError> {
-        Ok(self
-            .create_bucket(bucket.id.as_str(), bucket._type.as_str())
-            .unwrap())
+        self.create_bucket(bucket.id.as_str(), bucket._type.as_str())
+            .unwrap();
+        Ok(())
         //Err(DatastoreError::InternalError("Not implemented".to_string()))
     }
-    fn heartbeat(&self, bucket_id: &str, event: Event, duration: f64) -> Result<Event, String> {
-        Err("Not implemented".to_string())
+    fn heartbeat(&self, bucket_id: &str, event: Event, pulsetime: f64) -> Result<Event, String> {
+        self.heartbeat(bucket_id, &event, pulsetime).unwrap();
+        Ok(event)
+        //Err("Not implemented".to_string())
     }
 }
 
-/// Performs a single sync pass
+/// Performs a single sync pass, in one direction
+///
+/// Steps:
+///   - Check the remotes
+///   - Create any new buckets (avoiding bucket ID conflicts by appending '-synced-from-{deviceid}')
+///   - Importing any new events
+#[allow(dead_code)]
```

Comment:
Why allow this? Why not just comment the dead code out?

Reply:
It's not dead code, I don't know why it is reported as such.